client/components/data/query-simple-payments/index.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestProduct, requestProducts } from 'state/simple-payments/product-list/actions';
+
+class QuerySimplePayments extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			nextProps.siteId !== this.props.siteId ||
+			nextProps.productId !== this.props.productId
+		) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		const { siteId } = props;
+
+		if ( ! props.siteId ) {
+			return;
+		}
+
+		if ( props.allProducts ) {
```

Comment:
What if the `allProducts` props wasn't there at all, and I could query all products just by not specifying the `productId` prop? I see two advantages:
- `<QuerySimplePayments siteId={ 1234 } />` does something instead of nothing
- API is similar to the `getSimplePayments` selector